### PR TITLE
DOC: Copy edits to mrml_overview.md

### PR DESCRIPTION
**Pull request by Leengit**
Make a few grammatical and syntactical changes, but no change in information.

### CODE REVIEW

**Review by lassoan — COMMENTED** [review]
Thanks a lot for the thorough review and fixes. Apart from a few inline comments the only change I would suggest is not to change `3D Slicer` to `3D&nbsp;Slicer`. Breaking the line between these two words is not an issue, but making the words less readable is.

**Review by lassoan — COMMENTED** [review]
Thanks a lot for the thorough review and fixes. Apart from a few inline comments the only change I would suggest is not to change `3D Slicer` to `3D&nbsp;Slicer`. Breaking the line between these two words is not an issue, but making the words less readable is.

File: Docs/developer_guide/mrml_overview.md [hunk 1]
```diff
@@ -4,107 +4,107 @@
...
+  - **Model** ([vtkMRMLModelNode](https://apidocs.slicer.org/main/classvtkMRMLModelNode.html)): stores a surface mesh (polygonal elements, points, lines, etc.) or a volumetric mesh (tetrahedral, wedge elements, unstructured grid, etc.).
+  - **Segmentation** ([vtkMRMLSegmentationNode](https://apidocs.slicer.org/main/classvtkMRMLSegmentationNode.html)): complex data node that can store a image segmentation (also known as contouring, labeling). It can store multiple representations internally; for example it can store both a binary labelmap image and a closed surface mesh.
```
[the remark this answers is not in the document]
Comment (Leengit) [reply Docs/developer_guide/mrml_overview.md:24]: Will do.

File: Docs/developer_guide/mrml_overview.md [hunk 2]
```diff
@@ -4,107 +4,107 @@
...
 
-MRML Scene provides Undo/Redo mechanism that restores a previous state of the scene and individual nodes. By default, undo/redo is disabled and not displayed on the user interface, as it increased memory usage and was not tested thoroughly.
+MRML Scene provides an Undo/Redo mechanism that restores a previous state of the scene and individual nodes. By default, undo/redo is disabled and not displayed on the user interface, because it increased memory usage and was not tested thoroughly.
 
 Basic mechanism:
 - Undo/redo is based on saving and restoring the state of MRML nodes in the Scene.
-- MRML scene can save snapshot of all nodes into a special Undo and Redo stacks.
-- The Undo and Redo stacks store copies of nodes that have changed from the previous snapshot. The node that have not changed are stored by a reference (pointer).
-- When an Undo is called on the scene, the current state of Undo stack is copied into the current scene and also into Redo stack.
+- An MRML scene can save a snapshot of all nodes into special Undo and Redo stacks.
```
[the remark this answers is not in the document]
Comment (Leengit) [reply Docs/developer_guide/mrml_overview.md:89]: I'll go with "a MRML".

**Pull request merged**